### ISSUE TITLE
💅 lint/suspicious/noEmptyInterface false positive

### ISSUE THREAD

**Issue by jer-sen** [biomejs/biome#1157]
### Environment information

```bash
CLI:
  Version:                      1.4.1
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           windows

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.10.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "yarn/1.22.21"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### Rule name

lint/suspicious/noEmptyInterface

### Playground link

https://biomejs.dev/playground/?lintRules=all&code=aQBuAHQAZQByAGYAYQBjAGUAIABCACAAewAKACAAIABhADoAIABiAG8AbwBsAGUAYQBuACwACgAgACAAYgA6ACAAYgBvAG8AbABlAGEAbgAsAAoAfQAKAGkAbgB0AGUAcgBmAGEAYwBlACAAQQAgAGUAeAB0AGUAbgBkAHMAIABQAGkAYwBrADwAQgAsACAAJwBhACcAPgAgAHsAfQA%3D

### Expected result

Should not report an error when extending something since the result is not an empty interface.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by Conaclos** [comment]
`noEmptyInterface` also report empty interfaces that extend a type as showed in the diagnostic:

```
main.tsx:5:1 [lint/suspicious/noEmptyInterface](https://biomejs.dev/linter/rules/no-empty-interface)  FIXABLE  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ An interface declaring no members is equivalent to its supertype.
  
    3 │   b: boolean,
    4 │ }
  > 5 │ interface A extends Pick<B, 'a'> {}
      │ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  
  ℹ Safe fix: Convert empty interface to type alias.
  
    3 3 │     b: boolean,
    4 4 │   }
    5   │ - interface·A·extends·Pick<B,·'a'>·{}
      5 │ + type·A·=·Pick<B,·'a'>
```

**Comment by jer-sen** [comment]
What's your point?

Type alias is not the same than an interface.
Cf https://www.sitepoint.com/typescript-type-vs-interface/

**Comment by jer-sen** [comment]
We try to always use interfaces cf https://typescript-eslint.io/rules/consistent-type-definitions
For consistency, performance and inconsistency detection when declaring interface (instead of getting a `never` type detected where it is used).

**Comment by jer-sen** [comment]
You can have a look at this: https://github.com/microsoft/TypeScript/wiki/Performance#preferring-interfaces-over-intersections

**Comment by Conaclos** [comment]
Thanks for sharing this resource!

They recommend using interfaces that extend two or more types instead of using type intersections. Thus, this doesn't apply to empty interfaces that **extend a single type**. Although not epxlicitly said, I think the TypeScript team recommend using type aliases instead of interfaces when the interface is empty and extends a single type. This is why the TypeScript ESLint rule `no-empty-interface` only recommend removing empty interface that extends zero or a single type.

If you are always using interfaces over type aliases, I think you should disable the `noEmptyInterface` rule?

**Comment by blutorange** [comment]
Another case where you must use an interface: When you need to augment an interface.  With the infamous JQuery as an example:

```ts
declare global {
  interface JQuery extends JQueryFunctionExtensionsMyAwesomePlugin {}
}
```

The linter would rewrite the above as

```ts
declare global {
  type JQuery = JQueryFunctionExtensionsMyAwesomePlugin;
}
```

which does not augment the module, and is also a compiler error.

**Comment by Conaclos** [comment]
@blutorange
I recently fixed this for external module (See #959). Unfortunately I didn't think about global declaration :/

I opened #1243 to allow empty interfaces that extend a type in global declarations.

**Comment by blutorange** [comment]
@Conaclos  That sounds like a good idea, and should fix that use case, thanks for the PR : )

Out of curiosity, how do you feel about allowing empty interfaces that extend something in general, not only in module / global declarations? Code such as this is possible, albeit I'd hope less common:

```ts
interface Foo {
  x: () => void;
}

interface Bar {
  y: () => void;
}

interface Baz {
  z: () => void;
}

interface Foo extends Bar {}
```

One could argue that the interface is not empty when properties are pulled in via `extends`. It might also be a bit more understandable to users, since no error is reported if the number of extended interface is `> 1`:

```ts
// lint error (why do I have to change this to "type Foo = Bar" ?)
interface Foo extends Bar {}

// no lint error (why do I not have to change this to "type Foo = Bar & Baz" ?)
interface Foo extends Bar, Baz {}
```

**Comment by Conaclos** [comment]
Yes, I have thought about this change. However, I am a bit hesitant about it because it will be a divergence from TypeScript ESLint. We could wait a bit to get more feedback.

**Comment by jer-sen** [comment]
I think the (not obvious) choice between type/interface should be handled by `@typescript-eslint/consistent-type-definitions` .
The `noEmptyInterface` should prevent use of `{}` when the developer wants an "empty" object.

Some other interesting content:
- https://www.typescriptlang.org/play?q=291#example/types-vs-interfaces
- https://stackoverflow.com/a/65948871

**Comment by Conaclos** [comment]
The rule is now ignoring interfaces that extend a type. Thanks for your valuable comments!
